src/ChargeDrift/ChargeDrift.jl

```diff
@@ -335,12 +321,13 @@ function _drift_charge!(
 
     # if the drifting model has a calculate_mobility function (scalar mobility), then we use the mobility-tied diffusion model
     # TODO: adapt the code to 3D-vector mobility
-    use_mobility_tied_diffusion = :calculate_mobility in propertynames(det.semiconductor.charge_drift_model) 
+    use_mobility_tied_diffusion = hasproperty(det.semiconductor.charge_drift_model, :calculate_mobility)
     
     @inbounds for istep in 2:max_nsteps
         last_real_step_index += 1
         _set_to_zero_vector!(step_vectors)
-        _add_fieldvector_drift!(step_vectors, current_pos, done, electric_field, det, S)
+        killdrift_while_nofield && _add_fieldvector_drift!(step_vectors, current_pos, done, electric_field, det, S)
+        !killdrift_while_nofield && _add_fieldvector_drift_nokill_when_nofield!(step_vectors, current_pos, done, electric_field, det, S)

```

Comment:
Then, you could just pass this keyword argument to the function, e.g. using
```suggestion
_add_fieldvector_drift!(step_vectors, current_pos, done, electric_field, det, S, killdrift_while_nofield)
```

Reply:
I've thought this. This way is simpler, but N times slower then defining a new function (N is the carrier number). 

But I guess the if-else takes almost no time. I will change this.